### ISSUE TITLE
Batch normalization mini-batch stats and moving averages 

### ISSUE THREAD

**Issue by rauar** [thtrieu/darkflow#768]
Hi,

I'm by far not experienced enough with Tensorflow nor Neural Nets but my my research on effects based on different batch sizes I stumbled upon the topic that Tensorflow's batch normalization requires often explicit configuration to correctly work. However I could not find anything like that in darkflow.

A quick summary of what I'm talking about is here: http://davideng.me/2017/10/10/batch-normalization-a-bug-hidden-in-plain-sight.html

Copied from the link:

> Tldr;
> Batch normalization requires the distribution of each mini-batch to reflect the distribution of the entire dataset.
> Batch norm uses the mini-batch stats during training but the moving averages found in training during evaluation.
> In TensorFlow, batch norm requires its moving averages to be updated separately from γ and β. This requires the operator used for training to be wrapped in a call to tf.control_dependencies(update_ops).

I could not find tf.GraphKeys.UPDATE_OPS in the code so I'm wondering if it would be necessary for the YOLO models to correctly work.

Any comments ?
